app/scripts/directives/createSecret.js

```diff
@@ -143,6 +143,17 @@ angular.module("openshiftConsole")
         };
 
         var linkSecretToServiceAccount = function(secret) {
+          if (!$scope.serviceAccountsNames.contains($scope.newSecret.pickedServiceAccountToLink)) {
+            var alert = {
+              name: 'createAndLink',
+              data: {
+                type: "success",
+                message: "Secret " + secret.metadata.name + " was created but couldn't be linked with service account " + $scope.newSecret.pickedServiceAccountToLink + "."
```

Comment:
i'm not suggesting it be two separate UI actions the user can take, i'm just suggesting you treat it as two steps internally and report error w/ finer granularity.